pkg/universe.dagger.io/CODING_STYLE.md

```diff
@@ -0,0 +1,203 @@
+# Package Coding Style
+
+## Public names
+
+Choose `PascalCase` for ”public“ definitions (importable outside of package).
+
+```cue
+// good
+#WriteFile: {}
+
+// bad
+#writeFile: {}
+#write_file: {}
+```
+
+Choose `camelCase` for "public" fields, instead of `snake_case` or `"kebab-case"`.
+
+```cue
+actions: {
+    // good
+    goLint: ...
+
+    // bad
+    go_lint: ...
+    "go-lint": ...
+}
+```
+
+## Private names
+
+Choose `_#camelCase` for private definitions, that should only be used in current package.
+
+```cue
+// good
+_#mergeStructs: { ... }
+
+// bad, starts with lower case but not private
+#mergeStructs: { ... }
+
+// bad, starts with upper case but private
+_#MergeStructs: { ... }
+```
+
+Choose `_camelCase` for private fields (implementation details).
+
+```cue
+#Copy: {
+    // good, indicates internal implementation
+    _copy: core.#Copy
+
+    // bad, indicates it can be used directly
+    copy: core.#Copy
+}
+```
+
+## Definitions (`#`) for *schemas*, fields for concrete *implementations*
+
+```cue
+// good, defines a schema
+#Copy: {
+    // good, implements a schema
+    input: #Image
+    _copy: core.#Copy & { ... }
+}
+
+// bad, not a definition
+Copy: {
+    // bad, not using a visible or hidden field (according to intent)
+    #input: #Image
+    _#copy: core.#Copy & { ... }
+}
+
+// ok for mixin
+Mixin: { ... }
+copy: #Copy & Mixin & { ... }
+```
+
+## All public definitions and fields must be documented
+
+Other fields are optional. The main idea is that public fields should be documented for someone
```

Comment:
"Documenting other fields is optional". (otherwise it reads like you're talking about cue optional fields :)

Reply:
I see it 🙂